### PR TITLE
SONARGRADL-89 Add java 17 in CI to sonar-scanner-gradle

### PR DESCRIPTION
**Pull request by zipeng-wu-sonarsource**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -69,13 +69,13 @@ repositories {
 }
 
 dependencies {
-  implementation 'org.sonarsource.scanner.api:sonar-scanner-api:2.16.1.361'
+  implementation 'org.sonarsource.scanner.api:sonar-scanner-api:2.16.2.588'
```
Comment (dbmeneses) [inline build.gradle:72]: this is potentially a functional change, doesn't affect tests only.
Was it necessary for it to run with java 17?
If not, it should be done with a dedicated commit/ticket.

File: integrationTests/src/test/java/org/sonarqube/gradle/AbstractGradleIT.java [hunk 2]
```diff
@@ -129,7 +129,7 @@ protected RunResult runGradlewWithEnvQuietly(String project, String exeRelativeP
       .directory(exeDir)
       .redirectOutput(outputFile)
       .redirectErrorStream(true);
-    pb.environment().put("GRADLE_OPTS", "-Xmx1024m");
+    pb.environment().put("GRADLE_OPTS", "-Xmx1024m --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED");
```
Comment (dbmeneses) [inline integrationTests/src/test/java/org/sonarqube/gradle/AbstractGradleIT.java:132]: Does it mean that to run the scanner with java 17 we need to add this JVM parameter?
What fails if this is not passed?

Comment (zipeng-wu-sonarsource) [reply integrationTests/src/test/java/org/sonarqube/gradle/AbstractGradleIT.java:132]: It had this issue with Java 17(and gradle 7.3+ doesn't fix it): https://github.com/gradle/gradle/issues/18508

File: build.gradle (hunk 1 above)
Comment (zipeng-wu-sonarsource) [reply build.gradle:72]: yeah indeed, this update might not be neccessary, i will roll it back

**Comment by dbmeneses** [comment]
@zipeng-wu-sonarsource The commit ticket number/message should be updated to https://jira.sonarsource.com/browse/SONARGRADL-89

**Review by dbmeneses — APPROVED** [review]
great job 👍
